### ISSUE TITLE
Support .browserslistrc

### ISSUE THREAD

**Issue by Undistraction** [gatsbyjs/gatsby#24575]
## Summary

Browserslist is used by Gatsby and supports definition of supported browsers using a `.browserslistrc` file. However Gatsby ignores such a file if present in the root of a project and only accepts config for browerslist as an attribute in the project's `package.json`.

From the [docs](https://github.com/browserslist/browserslist):

All tools will find target browsers automatically, when you add the following to package.json:
```
  "browserslist": [
    "defaults",
    "not IE 11",
    "not IE_Mob 11",
    "maintained node versions"
  ]
```

Or in .browserslistrc config:

```
defaults
not IE 11
not IE_Mob 11
maintained node versions
```

### Motivation

Using a separate dotfile is much clearer and easier to generate, track and share between projects. Editing the list of browsers doesn't result in a change to `package.json`

**Comment by balibebas** [comment]
**TL;DR** [`.browserslistrc`](https://github.com/browserslist/browserslist#browserslistrc) is already supported but only in project root.

If you landed here because the docs don't mention use of the `.browserslistrc` file here's a tip: yes it works, but only in the root directory. To confirm create a `.browserslistrc` file in your project root, add some bad config then run `gatsby build`. You will see an error. Then fix the config. The error will not occur. I'm imagining that's what OP did before they closed this issue.

**Comment by balibebas** [comment]
**TL;DR** [`.browserslistrc`](https://github.com/browserslist/browserslist#browserslistrc) is already supported but only in project root.

If you landed here because the docs don't mention use of the `.browserslistrc` file here's a tip: yes it works, but only in the root directory. To confirm create a `.browserslistrc` file in your project root, add some bad config for the environment you wish to test then run `gatsby build`. You will see an error:

![Screen Capture_select-area_20200712192741](https://user-images.githubusercontent.com/65664227/87245181-eaef7200-c475-11ea-9ca5-1bdffaa59387.png)

Then fix the config. The error will no longer occur.
